### PR TITLE
Added assertion that checks for unique requests' names in the collection

### PR DESCRIPTION
**Pull request by PA-Rweins**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/42278

## Description
The requests are being checked for having unique names, if not integration will not be generated and a proper message will be shown.

## Must have
- [x] Tests
- [x] Documentation
- [ ] [VS Code Extension](https://docs.gitlab.com/ee/ci/yaml/#tags)
  - [ ] Functionality implemented in the extension - <u>\<link-to-pr-here\></u>
  - [ ] N/A - Functionality cannot be implemented in the extension because <u>\<add-reason-here\></u>

### CODE REVIEW

File: demisto_sdk/commands/postman_codegen/postman_codegen.py [hunk 1]
```diff
@@ -423,3 +425,27 @@ def generate_command_outputs(body: Union[Dict, List]) -> List[IntegrationGenerat
...
+
+def duplicate_requests_check(commands_names_dict: dict) -> None:
+    duplicates_list = []
+    for key in commands_names_dict:
+        if len(commands_names_dict[key]) > 1:
+            duplicates_list.extend(commands_names_dict[key])
+
+    assert len(duplicates_list) == 0, f'There are requests with non-unique names:{duplicates_list}.' \
+                                      f' \n you should give a unique name to each request. ' \
+                                      f' names are not case sensitive and whitespaces are ignored.'
```
Comment (Bargenish) [inline demisto_sdk/commands/postman_codegen/postman_codegen.py:451]:
```suggestion
    assert len(duplicates_list) == 0, f'There are requests with non-unique names: {duplicates_list}.\n' \
                                      f'You should give a unique name to each request.\n' \
                                      f'Names are case-insensitive and whitespaces are ignored.'
```

Comment (PA-Rweins) [reply demisto_sdk/commands/postman_codegen/postman_codegen.py:451]: Fixed

File: demisto_sdk/commands/postman_codegen/postman_codegen.py [hunk 2]
```diff
@@ -423,3 +425,27 @@ def generate_command_outputs(body: Union[Dict, List]) -> List[IntegrationGenerat
...
+def build_commands_names_dict(items: list) -> dict:
+    names_dict = {}
+    for item in items:
+        request_name = item.get('name', None)
+        if request_name:
+            command_name = tools.to_kebab_case(request_name)
+            if command_name not in names_dict:
+                names_dict[command_name] = [request_name]
+            else:
+                names_dict[command_name].append(request_name)
```
[the remark this answers is not in the document]
Comment (PA-Rweins) [reply demisto_sdk/commands/postman_codegen/postman_codegen.py:439]: Will definitely use it, thanks!

[the remark this answers is not in the document]
Comment (PA-Rweins) [reply demisto_sdk/commands/postman_codegen/postman_codegen.py]: fixed

**Review by Bargenish — APPROVED** [review]

**Comment by Bargenish** [comment]
Add to CHANGELOG as well pls.

**Pull request merged**